### DISCUSSION TITLE
git push fails: Connection closed by 20.205.243.166 port 22

### DISCUSSION THREAD

**Discussion by fzyzcjy** [community/community#54326] [Repositories]
### Select Topic Area

Bug

### Body

I cannot work because cannot push code :( (I need the code to be pushed and linked to issues etc...) I remembered a few months (?) ago, there was another similar incident.

Is this a failure of GitHub, or just my network problem?

```
git push --verbose
Pushing to github.com:fzyzcjy/myreponame.git
kex_exchange_identification: Connection closed by remote host
Connection closed by 20.205.243.166 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

also tried:

```
git pull
kex_exchange_identification: Connection closed by remote host
Connection closed by 20.205.243.166 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

network seems ok:

```
curl google.com
<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="http://www.google.com/">here</A>.
</BODY></HTML>
```

other repo https://github.com/fzyzcjy/flutter_rust_bridge also fails:

```
(base) ➜  flutter_rust_bridge git:(master) ✗ git pull
kex_exchange_identification: Connection closed by remote host
Connection closed by 20.205.243.166 port 22
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by fzyzcjy** [comment]
By switching from git to https it temporarily works.

**Reply by 769344359** [reply]
the same as you

**Reply by selfboot** [reply]
same problem

**Reply by dvosscricut** [reply]
same. oy vey!

**Reply by dvosscricut** [reply]
it's intermittent, If I do it on command line over and over after 30 tries or so it can get through and work. Seems to randomly not work and then work, with no other changes, just trying again later.

**Comment by Leenuus** [comment]
I fixed this problem following [this guide _Using SSH over the HTTPS port_ provided by Github](https://docs.github.com/en/authentication/troubleshooting-ssh/using-ssh-over-the-https-port). Strange network issue really...

**Reply by imann81** [reply]
thx!!

**Comment by MooMooHorse** [comment]
If you use proxy to connect to github by some clients like clash / v2ray, then you can do it like this [link](https://moomoohorse.github.io/blog/gitconnectionv2ray).

**Reply by cai-q** [reply]
Thanks, that really solved my problem.

**Reply by CarefreeState** [reply]
Thanks, bro😭

**Reply by leylatop** [reply]
Thans, very!

**Reply by smallzhong** [reply]
Thanks, man. You really saved me😭
